### PR TITLE
Updates included build tool to match the current 6.0.200 sdk

### PR DESCRIPTION
**Pull request by JoeRobich**
Includes the changes to completion and block structure options from https://github.com/OmniSharp/omnisharp-roslyn/pull/2306

### CODE REVIEW

**Review by filipw — APPROVED** [review]
thanks!

**Comment by filipw** [comment]
> Microsoft.Build.Exceptions.InvalidProjectFileException: The imported project "C:\Users\runneradmin\AppData\Local\Temp\541ofin3.s03\ProjectWithAnalyzers\obj\ProjectWithAnalyzers.csproj.*.props" was not found. Confirm that the expression in the Import declaration "C:\Users\runneradmin\AppData\Local\Temp\541ofin3.s03\ProjectWithAnalyzers\obj\ProjectWithAnalyzers.csproj.*.props" is correct, and that the file exists on disk.  D:\a\omnisharp-roslyn\omnisharp-roslyn\.msbuild\Current\Microsoft.Common.props

is this `*.props` some new MSBuild feature? I find it hard to comprehend what fails here
  
>  System.BadImageFormatException: Could not resolve field token 0x0400110f, due to: Could not load type of field 'Microsoft.Build.Shared.FileSystem.ManagedFileSystem+<>c:<>9__10_0' (1) due to: Could not load file or assembly 'Microsoft.IO.Redist, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. assembly:/Users/runner/work/omnisharp-roslyn/omnisharp-roslyn/bin/Release/OmniSharp.MSBuild.Tests/net472/Microsoft.Build.dll type:<>c member:(null)

Looks like on Unix we need to include `Microsoft.IO.Redist` in the local msbuild?

File: src/OmniSharp.Host/MSBuild/Discovery/MSBuildLocator.cs [hunk 1]
```diff
@@ -70,6 +70,17 @@ public void RegisterInstance(MSBuildInstance instance)
...
 
+            if (instance.DiscoveryType == DiscoveryType.StandAlone)
+            {
+                // MSBuild began relying on the Microsoft.IO.Redist library which does not work
+                // on MacOS or Linux platforms. Disabling the 17.0 feature wave opts us in to the
+                // earlier behavior of using System.IO.
+
+                // This issue only affects the MSBuild tools shipped with O# as Mono does not ship
+                // with a version of MSBuild that includes this change.
+                Environment.SetEnvironmentVariable("MSBUILDDISABLEFEATURESFROMVERSION", "17.0");
```
Comment (filipw) [inline src/OmniSharp.Host/MSBuild/Discovery/MSBuildLocator.cs:81]: this is some dark magic man

I think we need to expedite getting rid of the standalone msbuild feature as we discussed.

Comment (JoeRobich) [reply src/OmniSharp.Host/MSBuild/Discovery/MSBuildLocator.cs:81]: I agree. I opened an issue and it should get resolved for 17.2 (https://github.com/dotnet/msbuild/issues/7335), but not shipping is best path forward.

**Pull request merged**